local-cli/runAndroid/runAndroid.js

```diff
@@ -123,15 +123,28 @@ function buildAndRun(args, reject) {
 }
 
 function startServerInNewWindow() {
+  var yargV = require('yargs').argv;
+
   const launchPackagerScript = path.resolve(
     __dirname, '..', '..', 'packager', 'launchPackager.command'
   );
 
   if (process.platform === 'darwin') {
-    child_process.spawnSync('open', [launchPackagerScript]);
+    if (!yargV.open) {
+      return child_process.spawnSync('open', [launchPackagerScript]);
+    }
+    return child_process.spawnSync('open', ['-a', yargV.open, launchPackagerScript]);
+
   } else if (process.platform === 'linux') {
-    child_process.spawn(
-      'xterm',
+    if (!yargV.open){
+      return child_process.spawn(
+        'xterm',
+        ['-e', 'sh', launchPackagerScript],
+        {detached: true}
+      );
+    }
+    return child_process.spawn(
+      yargV.open,
```

Comment:
Looks like this should work although I don't have a Linux machine to test it either :)

Same nit as above.